foreman/client.py

```diff
@@ -21,6 +21,14 @@
 except ImportError:
     PLUGINS = []
 
+try:
+    from requests.packages.urllib3.util.retry import Retry
+    RETRIES = 5
+    DEFAULT_SLEEP = 0.5
+    with_retry = True
+except ImportError:
+    with_retry = False
+
```

Comment:
I suggest to leave the urllib3 functionality. From the research I am doing now it is the superior Retry functionality that exists so far. At least in the libraries we depend now. We really want to be able to control both the number or retries and sleep factor. Also the requests library depend on it heavily so I think we do not need that try/catch block for import at all. requests library is in our requirements.txt:

https://kev.inburke.com/kevin/urllib3-retries/

I also suggest to set at least a backoff factor. By default there is none. We better have retries 
to be exponentially delayed:

https://www.awsarchitectureblog.com/2015/03/backoff.html